image/types/src/lib.rs

```diff
@@ -392,4 +392,13 @@ impl ImageTocEntry {
     pub fn image_size(&self) -> u32 {
         self.size
     }
+
+    pub fn overlaps(&self, other: &ImageTocEntry) -> bool {
```

Comment:
The check here is incomplete. This only works if the param passed in has a higher load address than "self". For completeness, it would be best to check both directions.

Reply:
This is the standard overlap detection logic. Equivalent to:

```
a.start <= b.end && a.end >= b.start
```

(but I would argue it should be > instead of >=, and < instead of <=)